community/bolt/src/main/java/org/neo4j/bolt/runtime/DefaultBoltConnection.java

```diff
@@ -147,6 +147,7 @@ protected boolean processNextBatch( int batchCount )
     {
         try
         {
+            boolean expectOneMessage = batchCount == 1;
```

Comment:
1 looks like a magic parameter value now. Wdyt about adding another method parameter like `boolean exitIfNoJobsImmediatelyAvailable` and checking it instead of the `batchCount`?